packages/image_picker/ios/Tests/ImagePickerPluginTests.m

```diff
@@ -61,6 +61,22 @@ - (void)testPluginPickVideoDeviceBack {
                  UIImagePickerControllerCameraDeviceRear);
 }
 
+- (void)testPluginPickImageDeviceCancel {
+  if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+    return;
+  }
+  FLTImagePickerPlugin *plugin =
+      [[FLTImagePickerPlugin alloc] initWithViewController:[UIViewController new]];
+  FlutterMethodCall *call =
+      [FlutterMethodCall methodCallWithMethodName:@"pickImage"
+                                        arguments:@{@"source" : @(0), @"cameraDevice" : @(1)}];
+  [plugin handleMethodCall:call
+                    result:^(id _Nullable r){
+                    }];
+  // todo test imagePickerControllerDidCancel when result is nil
```

Comment:
I think we can just call two times of `imagePickerControllerDidCancel` here to simulate crash. @cyanglaz What do you think?

Reply:
I see #2688 fixed this, so need to close this PR?

Reply:
> I see #2688 fixed this, so need to close this PR?

Hey, you can rebase and add the test, we fix different crash. :)